### PR TITLE
Friendly error message when using stack with old Cabal library

### PR DESCRIPTION
**Pull request by mgsloan**
PR #4678 broke compatibility with Cabal <= 1.19.2

Repro is to `stack new old-test` then put `resolver: lts-2.0` in `stack.yaml`.  Result is:

```
mgsloan@treetop:~/oss/stack/old-test$ stack build
old-test-0.1.0.0: configure (lib + exe)
unrecognized option `--exact-configuration'

--  While building package old-test-0.1.0.0 using:
      /home/mgsloan/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_1.18.1.5_ghc-7.8.4 --builddir=.stack-work/dist/x86_64-linux/Cabal-1.18.1.5 configure --user --package-db=clear --package-db=global --package-db=/home/mgsloan/.stack/snapshots/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/pkgdb --package-db=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/pkgdb --libdir=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/lib --bindir=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/bin --datadir=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/share --libexecdir=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/libexec --sysconfdir=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/etc --docdir=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/doc/old-test-0.1.0.0 --htmldir=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/doc/old-test-0.1.0.0 --haddockdir=/home/mgsloan/oss/stack/old-test/.stack-work/install/x86_64-linux/2ed822369d342fd9bc33f97d40802cbf34243df8b0cf5e7288d1cd1bfeb1c696/7.8.4/doc/old-test-0.1.0.0 --constraint=base==4.7.0.2 --exact-configuration
    Process exited with code: ExitFailure 1
```

After merge of this PR result is:

```
mgsloan@treetop:~/oss/stack/old-test$ stack build
Stack no longer supports Cabal versions earlier than 1.19.2, but version 1.18.1.5 was found.  Consider updating the resolver to lts-3.0 or later / nightly-2015-05-05 or later.
```

* [X] Any changes that could be relevant to users have been recorded in the ChangeLog.md
* [X] The documentation has been updated, if necessary.

### CODE REVIEW

File: src/Stack/Build.hs [hunk 1]
```diff
@@ -121,6 +121,11 @@ checkCabalVersion = do
             "Error: --allow-newer requires at least Cabal version 1.22, but version " ++
             versionString cabalVer ++
             " was found."
+    when (cabalVer <= mkVersion [1, 19, 2]) $ throwM $
```
Comment (qrilka) [inline src/Stack/Build.hs]: Could you explain where does 1.19.2 come from? I don't even find that version on Hackage.
BTW #4475 talks about dropping Cabal<2 though it's just a discussion yet.

Comment (mgsloan) [reply src/Stack/Build.hs]: Ah right meant to link to, this comment in Cabal's code: https://github.com/haskell/cabal/blob/580fe6b6bf4e1648b2f66c1cb9da9f1f1378492c/cabal-install/Distribution/Client/Setup.hs#L592 .  I've added a comment including this link

**Comment by mgsloan** [comment]
@qrilka Note I've force pushed a squashed version.  Notable change from prior version is using `<` for the version comparison instead of `<=`

**Comment by snoyberg** [comment]
We have a few related issues:

* The current incompatibility on master
* #4246 
* #4475 

I'd rather see a change that addresses all of these, by:

* Warning on versions of GHC and Cabal that are too old
    * When appropriate, mention that an older version of Stack can be used instead
* Warning on untested versions of GHC and Cabal

**Comment by mgsloan** [comment]
Hmm, yes I suppose resolving #4246 and #4475 would be ideal, however those are both inactive and inconclusive discussions.

I figure that this change is an improvement over the state where old projects cause incorrect usage of Cabal.  Merging it would avoid the circumstance where nothing is done about this before the next release.

**Pull request closed without merging**